File: Euclid/Network/Streams/Util/ArgumentEntry.cs
namespace Euclid.Network.Streams.Util
{
    public class ArgumentEntry
    {
        #region Private variables

        private string m_Value;

        #endregion

        #region Fields

        public string Value
        {
            get { return m_Value; }
        }

        #endregion

        #region Constructor

        public ArgumentEntry(object value = null)
        {
            m_Value = (value != null ? value.ToString() : string.Empty);
        }

        #endregion
    }
}

File: euclid/src/main/java/euclid/network/streams/util/ArgumentEntry.java
package euclid.network.streams.util;

public final class ArgumentEntry {

    private final String value;

    public ArgumentEntry() {
        this.value = "";
    }

    public ArgumentEntry(Object value) {
        this.value = (value != null ? value.toString() : "");
    }

    public String getValue() {
        return value;
    }
}
